js/lib/src/state/namespace.ts

```diff
@@ -11,10 +11,10 @@ export class PermissivenessSettings {
 }
 
 export enum Permissiveness {
-  All,
-  Whitelist,
-  Blacklist,
-  Namespace,
+  All = "ALL",
```

Comment:
We're moving to an approach more along the lines of this https://github.com/raindrops-protocol/raindrops/blob/57b1f4a19923a9ff0b919f0373ea10648cd506dc/js/lib/src/state/item.ts#L305, with a `toInstruction` and `fromInstruction`(from hasn't been implemented yet) interface.

Reply:
This looks very interesting, I like the approach. I am hesitant to change this in this PR because yours is not merged yet so I would need to pull in your PR for the `Instructable` types in `common`

Reply:
You can just steal the `Instructable` interface from my PR and add it here. I'll deal with the conflicts if you get this PR merged before me.

Reply:
Id prefer to do this in a subsequent PR after yours has been merged and approved, its quite a large change and this PR feels pretty big as is, what do you think?